test/test_02_hw_05.py

```diff
@@ -0,0 +1,38 @@
+import functools
+
+from hw.hw_05_task_02 import print_result
+
+import pytest
+
+
+@print_result
+def custom_sum(*args):
+    """Description of the function."""
+    return functools.reduce(lambda x, y: x + y, args)
+
+
+def test__doc__():
+    assert custom_sum.__doc__ == "Description of the function."
+
+
+def test__name__():
+    assert custom_sum.__name__ == "custom_sum"
+
+
+def test_has_atr__oiginal_func():
+
+    assert hasattr(custom_sum, "__original_func") is True
+
+
+def func_without_wrap():
+    return 6
+
+
+def test__original_func():
+    with_print = print_result(func_without_wrap)
+    assert with_print.__original_func == func_without_wrap
```

Comment:
Declare test function inside the test. It is part of the test, no need to move it outside.  

Or you can use some existing function like sum.